dask/tests/test_base.py

```diff
@@ -485,7 +556,7 @@ class BDataClass:
 def test_tokenize_dataclass():
     a1 = ADataClass(1)
     a2 = ADataClass(2)
-    assert tokenize(a1) == tokenize(a1)
```

Comment:
This was actually broken and shadowed by the cache. Hence, the cache is cleared now. I corrected the dataclass implementation but unfortunately, I had to access some semi-private attributes. I think that it should be OK but this should be reviewed